### PR TITLE
Add test pipeline

### PR DESCRIPTION
**Pull request by abrauchli**
Check the following:

 - [ ] Breaking changes marked in commit message
 - [x] Changelog updated
 - [x] Code style cleaned (ran `make style-fix`)
 - [x] Tested on actual hardware

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+stages:
+  - build
+
+trigger_testbed:
+  stage: build
+  image:
+    name: registry.gitlab.sensirion.lokal/sensirion/docker/docker-ubuntu:18.04-1.0.0
```
Comment (rnestler) [inline .gitlab-ci.yml]: Can you bump to  18.04-1.2.0 ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
                 and CFLAGS
  * [`changed`]  Only one example with either `hw_i2c` or `sw_i2c` is built,
                 depending on `CONFIG_I2C_TYPE`. Defaults to `hw_i2c`.
+ * [`fixed`]  Measurement duration could take at most 15.5ms (datasheet) but we
+              only waited 15ms
+ * [`added`]  Add `STS3X_MEASUREMENT_DURATION_USEC` to header
```
Comment (rnestler) [inline CHANGELOG.md]: These changes have nothing to do with the test pipeline, have they?

Comment (abrauchli) [reply CHANGELOG.md]: Kinda: They have to do with the tests ran during the pipeline.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  image:
+    name: registry.gitlab.sensirion.lokal/sensirion/docker/docker-ubuntu:18.04-1.2.0
+  tags:
+    - docker
+    - linux
+  script:
+  - >
+    curl
+    --request POST
+    --form ref=test-pipeline
```
Comment (rnestler) [inline .gitlab-ci.yml]: Should be master right?

File: .gitlab-ci.yml (hunk 1 above)
Comment (abrauchli) [reply .gitlab-ci.yml]: Sure, fixed.

File: .gitlab-ci.yml (hunk 3 above)
Comment (abrauchli) [reply .gitlab-ci.yml]: yes, thanks. A leftover from the chicken and egg problem ;)

**Pull request merged**